### PR TITLE
feat: AIILG-463-implement REGARD scorer to evaluate model output bias and integ…

### PR DESCRIPTION
**Pull request by robbfox**
### Summary
This PR integrates the **REGARD** evaluation metric into the summarisation bias evaluation pipeline. REGARD is specifically designed to measure bias in generated text by scoring the sentiment and "regard" (positive, neutral, negative, other) towards different demographic groups in counterfactual scenarios.

### Key Changes
- **New REGARD Scorer**: Implemented the `REGARDScorer` in `evals/summarisation/src/bias/regard_scorer.py` to provide nuanced sentiment analysis beyond standard polarity.
- **Pipeline Integration**: 
    - Updated `runner.py` and `record_builder.py` to include REGARD scoring during counterfactual iterations.
    - Integrated REGARD metrics into the `output_formatter.py` to calculate mean scores and deltas across iterations.
- **Improved Bias Reporting**: Enhanced the plotting and summary output to display REGARD "negative" sentiment deltas, providing deeper insights into model bias.
- **Code Hygiene**: Renamed `evals/summarisation/src/bias/types.py` to `bias_types.py` to avoid potential naming conflicts and updated all associated imports.
- **Comprehensive Testing**: Added a robust suite of unit tests for the REGARD scorer (`test_regard_scorer.py`) and updated existing bias pipeline tests to accommodate the new metrics.

### Verification Results
Successfully ran the pipeline using counterfactual datasets, verifying that REGARD scores are correctly calculated, aggregated, and included in the final `results.jsonl` and `summary.json` outputs.

### CODE REVIEW

File: evals/summarisation/src/bias/regard_scorer.py [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+import torch
+from transformers import AutoModelForSequenceClassification, AutoTokenizer
+
+logger = logging.getLogger(__name__)
+
+# ---------------------------------------------------------------------------
+# Module-level constants
+# ---------------------------------------------------------------------------
+
+_MODEL_NAME: str = "sasha/regardv3"
```
Comment (marianasoftwire) [inline evals/summarisation/src/bias/regard_scorer.py:51]: Worth pinning sasha/regardv3 to a specific revision SHA via revision= on both AutoTokenizer.from_pretrained and AutoModelForSequenceClassification.from_pretrained. A bias-measurement tool with an unpinned model means a silent upstream update shifts the baseline of every run with no signal, and the version isn't reproducible from the lockfile alone.

Comment (robbfox) [reply evals/summarisation/src/bias/regard_scorer.py:51]: Ah okay - there only appear to be a small number of commits on Hugging Face for this model, all made on the same day in 2022. While it’s unlikely to change upstream, I’ve pinned the latest commit SHA to make the dependency explicit and ensure reproducibility and auditability.

Comment (marianasoftwire) [reply evals/summarisation/src/bias/regard_scorer.py:51]: Nice, Thanks.

**Review by marianasoftwire — APPROVED** [review]
LGTM
